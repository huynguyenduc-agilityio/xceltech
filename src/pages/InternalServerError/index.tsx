import { Link } from 'react-router-dom';

import { USER_PAGE } from '@/constants';

const InternalServerError = ({
  path = USER_PAGE.DASHBOARD,
}: {
  path?: string;
}) => {
  return (
    <div className="flex items-center justify-center py-[300px]">
      <div className="text-center">
        <h1 className="text-black-default text-2xl font-semibold">
          500 Internal Server Error
        </h1>
        <p className="text-lg font-medium text-muted-foreground mt-2 max-w-[414px] mx-auto">
          Sorry, something went wrong. :(
        </p>

        <div className="mt-10 h-[70px] text-white">
          <Link
            to={path}
            className="bg-primary rounded-regular px-6 py-3 mt-10 font-bold text-lg transition"
          >
            Back to Dashboard
          </Link>
        </div>
      </div>
    </div>
  );
};

export default InternalServerError;
