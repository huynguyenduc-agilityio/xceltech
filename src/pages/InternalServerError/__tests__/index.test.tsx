import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Pages
import InternalServerError from '..';

describe('InternalServerError Page', () => {
  test('should match snapshot', () => {
    const view = render(
      <BrowserRouter>
        <InternalServerError />
      </BrowserRouter>,
    );

    expect(view).toMatchSnapshot();
  });
});
