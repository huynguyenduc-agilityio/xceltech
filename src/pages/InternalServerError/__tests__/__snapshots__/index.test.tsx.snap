// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalServerError Page should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center py-[300px]"
      >
        <div
          class="text-center"
        >
          <h1
            class="text-black-default text-2xl font-semibold"
          >
            500 Internal Server Error
          </h1>
          <p
            class="text-lg font-medium text-muted-foreground mt-2 max-w-[414px] mx-auto"
          >
            Sorry, something went wrong. :(
          </p>
          <div
            class="mt-10 h-[70px] text-white"
          >
            <a
              class="bg-primary rounded-regular px-6 py-3 mt-10 font-bold text-lg transition"
              data-discover="true"
              href="/user/dashboard"
            >
              Back to Dashboard
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center py-[300px]"
    >
      <div
        class="text-center"
      >
        <h1
          class="text-black-default text-2xl font-semibold"
        >
          500 Internal Server Error
        </h1>
        <p
          class="text-lg font-medium text-muted-foreground mt-2 max-w-[414px] mx-auto"
        >
          Sorry, something went wrong. :(
        </p>
        <div
          class="mt-10 h-[70px] text-white"
        >
          <a
            class="bg-primary rounded-regular px-6 py-3 mt-10 font-bold text-lg transition"
            data-discover="true"
            href="/user/dashboard"
          >
            Back to Dashboard
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
